Inconsistency in help tab & small typo

NekoJonez opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

I described my changes in the Trac ticket. But the long and short of it is that I was checking some strings for the upcoming 6.1 release in my nl_BE translation and found some odd inconsistent things in the help tabs. With this PR, I want to fix them.

Trac ticket: (https://core.trac.wordpress.org/ticket/56921)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

NekoJonez committed:
Small consistency update in help section. 

Post list is mentioned in two different ways in the same sentence.

NekoJonez committed:
Small text improvements to the delete site mail.

During the translation check for 6.1, I felt that this mail didn't flow too well.

NekoJonez committed:
Consistency in help tab

Everything else starts with "Documentation on ..." and not auto-updates?

NekoJonez committed:
Consistency in help tab

Everything else starts with "Documentation on ..." and not auto-updates?

NekoJonez committed:
Consistency in help tab

Everything else starts with "Documentation on ..." and not auto-updates?

NekoJonez committed:
Consistency in help tab

Everything else starts with "Documentation on ..." and not auto-updates?

mukeshpanchal27 reviewed (approved):
Thanks @NekoJonez LGTM

audrasjb reviewed (approved):
Alright, the changes look good to me.

audrasjb committed:
Remove first change

As per Sergey's comment in the ticket: 

The suggested sentence repeats "post list" twice, and it's also "post list" in the first instance and "posts list" in the second:
> You can filter the post list by post status using the text links above the posts list to only show posts with that status. The default view is to show all posts.
The original sentence seems easier to read to me, and avoids unnecessary repetition:
> You can filter the list of posts by post status using the text links above the posts list to only show posts with that status. The default view is to show all posts.

audrasjb commented:
Fixed in https://core.trac.wordpress.org/changeset/54839 and https://core.trac.wordpress.org/changeset/54840.
I separated the patch in two commits to give them their own separate commit description.
